src/pygama/raw/orca/orca_digitizers.py

```diff
@@ -210,3 +204,228 @@ def decode_packet(self, packet, packet_id, rbl):
+            "energy_first": {
+                "dtype": "uint32",
+            },
+            "timestamp": {
+                "dtype": "uint64",
+                "units": "clock_ticks",
+            },
+            "crate": {
+                "dtype": "uint8",
+            },
+            "card": {
+                "dtype": "uint8",
+            },
+            "channel": {
+                "dtype": "uint8",
+            },
+            # waveform data
+            "waveform": {
+                "dtype": "uint16",
+                "datatype": "waveform",
+                "wf_len": 65532,  # max value. override this before initializing buffers to save RAM
+                "dt": 8,  # override if a different clock rate is used
+                "dt_units": "ns",
+                "t0_units": "ns",
+            },
+        }
+
+        self.decoded_values = {}
+        self.skipped_channels = {}
+        super().__init__(
+            header=header, **kwargs
+        )  # also initializes the garbage df (whatever that means...)
+
+    def set_header(self, header: OrcaHeader) -> None:
+        self.header = header
+
+        self.decoded_values = copy.deepcopy(self.decoded_values_template)
+
+        for card_dict in self.header["ObjectInfo"]["Crates"][0]["Cards"]:
+            if card_dict["Class Name"] == "ORSIS3316Model":
+                card = card_dict["Card"]
+                crate = 0
+                for channel in range(0, 16):
+                    ccc = get_ccc(crate, card, channel)
+                    trace_length = card_dict["rawDataBufferLen"]
+                    self.decoded_values[ccc] = copy.deepcopy(
+                        self.decoded_values_template
+                    )
+
+                    if trace_length <= 0 or trace_length > 2**16:
+                        raise RuntimeError(
+                            "invalid trace_length: ",
+                            trace_length,
+                        )
+
+                    self.decoded_values[ccc]["waveform"]["wf_len"] = trace_length
+
+        self.decoded_values["waveform"]["wf_len"] = trace_length
+
+    def get_key_list(self) -> list[int]:
+        key_list = []
+        for key in self.decoded_values.keys():
+            key_list += [key]
+        return key_list
+
+    def get_decoded_values(self, key: int = None) -> dict[str, Any]:
+        if key is None:
+            dec_vals_list = self.decoded_values
+            if len(dec_vals_list) == 0:
+                raise RuntimeError("decoded_values not built yet!")
+
+            return dec_vals_list  # Get first thing we find
+
+        if key in self.decoded_values:
+            dec_vals_list = self.decoded_values[key]
+            return dec_vals_list
+        raise RuntimeError("no decoded values for key", key)
+
+    def decode_packet(
+        self, packet: OrcaPacket, packet_id: int, rbl: RawBufferLibrary
+    ) -> bool:
+        """Decode the ORCA SIS3316 ADC packet.
+
+        The packet is formatted as:
+
```

Comment:
I couldn't find this card documented at
http://orca.physics.unc.edu/orca/VME/Index.html
Let's ask @tcald to add a page on it.

Reply:
I added those in my new commit along with fixing the above issues.